src/utilities.cpp

```diff
@@ -1167,4 +1168,53 @@ std::string replace(std::string string, const std::string &from, const std::stri
                string.replace(index, from.length(), to);
 }
 
+bool equalEntities(const EntityPtr &owner, const std::vector<EntityPtr> &entities)
+{
```

Comment:
Does this work when the order of entities is not matched?

Reply:
Yes, `unmatchedIndex` is used for this.

Reply:
Yes